packages/Tag/src/Tag.tsx

```diff
@@ -21,44 +26,92 @@ export type Appearance =
 export interface TagProps extends React.ComponentProps<'div'> {
   /** The different appearances of the Tag */
   appearance?: Appearance;
-  /** Add a data-test tag for automated tests */
-  dataTest?: string;
   /** The content of the Tag */
   children: React.ReactNode;
   /** Add a specific class to the Tag */
   className?: string;
+  /** Add a colored square instead of an image or an icon */
+  color?: string;
+  /** Add a data-test tag for automated tests */
+  dataTest?: string;
   /** Render the close button */
   dismissible?: boolean;
   /** The icon used at the beginning of the Tag */
-  icon?: React.ReactNode;
+  icon?: React.ReactElement;
+  /** The id of the tag */
+  id?: string;
+  /** Add an error style to the tag */
+  hasError?: boolean;
+  /** Add a warning icon to the tag. Overrides any image, color or icon */
+  isWarning?: boolean;
   /** Event when the tag is closed */
   onClose?: () => void;
   /** Render rounded corners */
   rounded?: boolean;
   /** The different sizes of the Tag */
   size?: Size;
+  /** Specifies the image url to show */
+  src?: string;
 }
 
 const Tag: React.FunctionComponent<TagProps> = ({
-  dataTest,
   children,
   className,
+  color,
+  dataTest,
   dismissible = false,
   appearance = 'default',
   icon,
+  id,
+  hasError,
+  isWarning,
   onClose,
   rounded = false,
   size = 'medium',
+  src,
 }) => {
   const [show, setShow] = React.useState(true);
 
-  const renderDismissButton = (
-    setShow: (show: boolean) => void,
-    onDismissClick?: () => void
-  ): JSX.Element => {
+  const renderIcon = (): JSX.Element | null => {
+    let visual = null;
+
+    if (isWarning) {
```

Comment:
To avoid if else if it is better to use conditional rendering.

```
    if (isWarning) {
      return (
        <IconAlert
          size="small"
          className="ids-tag__visual ids-tag__warning-icon"
        />
      );
    }

    if (icon) {
      return React.cloneElement(icon, {
        className: 'ids-tag__visual ids-tag__icon',
      });
    }

  ...

  return null
```